pkg/services/ngalert/notifier/channels/email.go

```diff
@@ -2,96 +2,104 @@ package channels
-	tmpl        *template.Template
+	log      channels.Logger
+	ns       channels.EmailSender
+	images   channels.ImageStore
+	tmpl     *template.Template
+	settings *emailSettings
 }
 
-type EmailConfig struct {
-	*channels.NotificationChannelConfig
+type emailSettings struct {
 	SingleEmail bool
 	Addresses   []string
 	Message     string
 	Subject     string
 }
 
 func EmailFactory(fc channels.FactoryConfig) (channels.NotificationChannel, error) {
-	cfg, err := NewEmailConfig(fc.Config)
+	notifier, err := buildEmailNotifier(fc)
 	if err != nil {
 		return nil, receiverInitError{
 			Reason: err.Error(),
 			Cfg:    *fc.Config,
 		}
 	}
-	return NewEmailNotifier(cfg, fc.Logger, fc.NotificationService, fc.ImageStore, fc.Template), nil
+	return notifier, nil
 }
 
-func NewEmailConfig(config *channels.NotificationChannelConfig) (*EmailConfig, error) {
-	settings, err := simplejson.NewJson(config.Settings)
+func buildEmailSettings(fc channels.FactoryConfig) (*emailSettings, error) {
+	type emailSettingsRaw struct {
+		SingleEmail bool   `json:"singleEmail,omitempty"`
+		Addresses   string `json:"addresses,omitempty"`
+		Message     string `json:"message,omitempty"`
+		Subject     string `json:"subject,omitempty"`
+	}
+
+	var settings emailSettingsRaw
+	err := json.Unmarshal(fc.Config.Settings, &settings)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to unmarshal settings: %w", err)
 	}
-	addressesString := settings.Get("addresses").MustString()
-	if addressesString == "" {
+	if settings.Addresses == "" {
 		return nil, errors.New("could not find addresses in settings")
 	}
 	// split addresses with a few different ways
-	addresses := util.SplitEmails(addressesString)
-	return &EmailConfig{
-		NotificationChannelConfig: config,
-		SingleEmail:               settings.Get("singleEmail").MustBool(false),
-		Message:                   settings.Get("message").MustString(),
-		Subject:                   settings.Get("subject").MustString(channels.DefaultMessageTitleEmbed),
-		Addresses:                 addresses,
+	addresses := splitEmails(settings.Addresses)
+
+	if settings.Subject == "" {
+		settings.Subject = channels.DefaultMessageTitleEmbed
+	}
+
+	return &emailSettings{
+		SingleEmail: settings.SingleEmail,
+		Message:     settings.Message,
+		Subject:     settings.Subject,
+		Addresses:   addresses,
 	}, nil
 }
 
 // NewEmailNotifier is the constructor function
 // for the EmailNotifier.
-func NewEmailNotifier(config *EmailConfig, l channels.Logger, ns channels.EmailSender, images channels.ImageStore, t *template.Template) *EmailNotifier {
-	return &EmailNotifier{
-		Base:        channels.NewBase(config.NotificationChannelConfig),
-		Addresses:   config.Addresses,
-		SingleEmail: config.SingleEmail,
-		Message:     config.Message,
-		Subject:     config.Subject,
-		log:         l,
-		ns:          ns,
-		images:      images,
-		tmpl:        t,
+func buildEmailNotifier(fc channels.FactoryConfig) (*EmailNotifier, error) {
```

Comment:
Either we change the name of this function to `newEmailNotifier` and un-capitalize the function name in the comment, or we update the comment above to mention the correct function name. I'm leaning towards the first option, but that's up to you.

Reply:
this follows the common pattern we chose when we started refactoring notifiers.